experiment_prototype/experiment_prototype.py

```diff
@@ -840,24 +967,38 @@ def edit_slice(self, edit_slice_id, **kwargs):
                 errmsg = 'Cannot edit slice: {} not a valid slice parameter'.format(edit_slice_param)
                 raise ExperimentException(errmsg)
 
-        # Move the interface values from old slice to new slice.
-        new_interface_values = {}
-        for ifkey, ifvalue in self._interface.items():
-            if edit_slice_id == ifkey[0]:
-                new_interface_values[ifkey[1]] = ifvalue
-            elif edit_slice_id == ifkey[1]:
-                new_interface_values[ifkey[0]] = ifvalue
+        # Get the interface values of the slice. These are not editable, if
+        # these are wished to be changed add_slice must be used explicitly
+        # to interface a new slice.
+        interface_values = self.get_slice_interfacing(edit_slice_id)
 
-        # This interface value will be removed when the slice is deleted.
-        new_interface_values[edit_slice_id] = None
+        removed_slice = self.del_slice(edit_slice_id)
```

Comment:
del_slice returns the slice dictionary that was removed, not a slice id, so it should be holding the slice information from the edit_slice_id here, and I don't think it will return anything else. This is just being held here in case there are errors in editing so we can revert. I don't think del_slice can return anything new